Engine/DataFeeds/LiveOptionChainProvider.cs

```diff
@@ -50,7 +54,27 @@ public IEnumerable<Symbol> GetOptionContractList(Symbol symbol, DateTime date)
                 throw new NotSupportedException($"LiveOptionChainProvider.GetOptionContractList(): SecurityType.Equity is expected but was {symbol.SecurityType}");
             }
 
-            return FindOptionContracts(symbol.Value);
+            var attempt = 1;
+            var contracts = Enumerable.Empty<Symbol>();
+
+            do
+            {
+                try
+                {
+                    Log.Trace($"LiveOptionChainProvider.GetOptionContractList(): Fetching option chain for {symbol.Value} [Attempt {attempt}]");
+
+                    contracts = FindOptionContracts(symbol.Value);
+                    break;
+                }
+                catch (Exception exception)
```

Comment:
Wondering if we want to catch any type of exception or maybe better just the web request one?
Also if we fail do download the `contracts` 5 times, wouldn't we want to finally throw?

Reply:
Good points! 
An exception which is not a `WebException` could be caused by updated content/format by OCC and should be reported immediately, not only logged silently.
If 5 download attempts fail, it should definitely throw.
Will fix both.